include/crocoddyl/multibody/frames.hpp

```diff
@@ -41,6 +41,11 @@ struct FrameTranslationTpl {
     return os;
   }
 
+  template <typename OtherScalar>
+  bool operator==(const FrameTranslationTpl<OtherScalar>& other) const {
+    return id == other.id && translation == other.translation.template cast<Scalar>();
```

Comment:
Should not be good to do a cast here.

Reply:
I forgot what type `translation` was. Thanks for the catch!

Reply:
ah no. It seems to be okay. Why is the cast not a good idea here?

Reply:
Because if cast a Scalar type to a larger type, then you will lose resolutions and things won't match.
The operator== should only be with the type `FrameTranslationTpl<Scalar>`